### PR TITLE
graphql: validate block params

### PR DESCRIPTION
**Pull request by s1na**
Block takes a `number` and a `hash`. The spec is unclear on this. My interpretation is either of these should be provided and we should error in case both are passed in.

Fixes https://github.com/ethereum/go-ethereum/issues/25899

### CODE REVIEW

File: graphql/graphql.go [hunk 1]
```diff
@@ -1207,6 +1207,9 @@ func (r *Resolver) Block(ctx context.Context, args struct {
 	Number *Long
 	Hash   *common.Hash
 }) (*Block, error) {
+	if args.Number != nil && args.Hash != nil {
+		return nil, errors.New("either number or hash must be specified")
```
Comment (holiman) [inline graphql/graphql.go:1211]: I don't think it's really right. It reads as if the error is that nothing has been specified. 
How about something more like these? I'm not super-happy about them either.. :/ 

```suggestion
		return nil, errors.New("only one of [number,hash] must be specified")
```
```suggestion
		return nil, errors.New("specify either number nor hash, not both")
```

Comment (s1na) [reply graphql/graphql.go]: I changed to `only one of number or hash must be specified`. Hopefully that's more clear

**Comment by wsb1994** [comment]
Hi, I personally disagree with the direction this goes -> Not because this is a bad idea, but because there is a language centric method of validation for things like this already in go-validator, here is an example of how that might look, although I'll allow you and the reviewer to decide on what that means for this PR and allow you to retain the edit :)!

`func (r *Resolver) Block(ctx context.Context, args struct {
	Number *Long
	Hash   *common
}) (*Block, error) {
	// Validate input arguments
	err := validateBlockArgs(args)
	if err != nil {
		return nil, err
	}
`




above is the validation function, using go-validate, which would then run the validation and return an error inline.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**